File: LeetCode75_LevelTwo/Happy_Number/Happy_Number.cc
// 202. Happy Number C++
#include <math.h>

class Solution {
public:
    bool isHappy(int n) {
        int slow_runner = n;
        int fast_runner = nextNum(n);
        while (fast_runner != 1 && slow_runner != fast_runner) {
            slow_runner = nextNum(slow_runner);
            fast_runner = nextNum(nextNum(fast_runner));
        }
        return fast_runner == 1;
    }
    int nextNum(int n) {
        int new_num = 0;
        while (n > 0) {
            new_num += pow(n%10, 2);
            n = n / 10;
        }
        return new_num;
    }
};

File: LeetCode75_LevelTwo/Happy_Number/Happy_Number.py
# 202. Happy Number Python

class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        hash_set = set()
        while n != 1 and n not in hash_set:
            hash_set.add(n)
            new_num = 0
            while n > 0:
                new_num += pow(n % 10, 2)
                n = n // 10
            n = new_num
        return n == 1
